vowpalwabbit/search.cc

```diff
@@ -301,6 +301,7 @@ search::~search()
     search_private& priv = *this->priv;
     clear_cache_hash_map(priv);
 
+    priv._random_state.~shared_ptr<rand_state>();
```

Comment:
Since _random_state is a shared_pointer<>, should it not properly delete itself via RAII? Why do you need to call the destructor directly?

Reply:
The destructor of priv is never called. It is deallocated using `free`. This is on the todo list of "make sure destructors are used"